backend/notifications/telegram.py

```
@@ -0,0 +1,457 @@
+                # Rate limited: unclaim this row and the rest of the batch,
+                # stop the run. The next cron tick resumes.
+                remaining_pks = [m.pk for m in batch[index:]]
+                TelegramMessage.objects.filter(pk__in=remaining_pks).update(
+                    status=TelegramMessage.STATUS_PENDING,
+                    attempts=F('attempts') - 1,
+                )
+                rate_limited = True
+                logger.warning("Telegram rate limited (retry_after=%s); run stopped", retry_after)
+                break
+            else:
+                if message.attempts >= MAX_ATTEMPTS:
```

Ruff (0.15.21)

[warning] Use `elif` instead of `else` then `if`, to reduce indentation

Convert to `elif`

(PLR5501)